File: src/Q21.cpp
#include <iostream>
#include <vector>
using namespace std;

class Solution {
public:
  vector<int> exchange(vector<int> &nums) {
    vector<int> odds, evens;
    for (auto num : nums) {
      if (num & 1) {
        odds.emplace_back(num);
      } else {
        evens.emplace_back(num);
      }
    }
    odds.insert(odds.end(), evens.begin(), evens.end());
    return odds;
  }
};

File: src/Q21.rs
struct Solution;

impl Solution {
    pub fn exchange(nums: Vec<i32>) -> Vec<i32> {
        let mut odds = Vec::new();
        let mut evens = Vec::new();
        for num in nums {
            if num & 1 == 1 {
                odds.push(num);
            } else {
                evens.push(num);
            }
        }
        odds.append(&mut evens);
        odds
    }
}
